src/native/corehost/fxr/fx_muxer.cpp

```diff
@@ -1018,21 +1019,68 @@ int fx_muxer_t::handle_exec_host_command(
         required_buffer_size);
 }
 
+namespace
+{
+    pal::architecture get_requested_architecture(int argc, const pal::char_t* argv[])
+    {
+        // Default to current architecture if architecture is not specified
+        if (argc < 3 || pal::strcasecmp(_X("--arch"), argv[2]) != 0)
+            return get_current_arch();
+
+        if (argc < 4)
+        {
+            trace::error(_X("Architecture should be specified when using --arch"));
+            return pal::architecture::__last;
+        }
+
+        pal::string_t arch_arg = argv[3];
+        for (int i = 0; i < static_cast<int>(pal::architecture::__last); ++i)
+        {
+            pal::architecture arch = static_cast<pal::architecture>(i);
+            if (pal::strcasecmp(get_arch_name(arch), arch_arg.c_str())  == 0)
+                return arch;
+        }
+
+        trace::error(_X("Unknown architecture: %s"), arch_arg.c_str());
+        return pal::architecture::__last;
+    }
+}
+
 int fx_muxer_t::handle_cli(
     const host_startup_info_t& host_info,
     int argc,
     const pal::char_t* argv[],
     const pal::string_t& app_candidate)
 {
     // Check for commands that don't depend on the CLI SDK to be loaded
-    if (pal::strcasecmp(_X("--list-sdks"), argv[1]) == 0)
-    {
-        sdk_info::print_all_sdks(host_info.dotnet_root, _X(""));
-        return StatusCode::Success;
-    }
-    else if (pal::strcasecmp(_X("--list-runtimes"), argv[1]) == 0)
+    bool list_sdks = pal::strcasecmp(_X("--list-sdks"), argv[1]) == 0;
+    bool list_runtimes = !list_sdks && pal::strcasecmp(_X("--list-runtimes"), argv[1]) == 0;
+
+    if (list_sdks || list_runtimes)
     {
-        framework_info::print_all_frameworks(host_info.dotnet_root, _X(""));
+        pal::architecture arch = get_requested_architecture(argc, argv);
```

Comment:
```suggestion
        pal::architecture arch = get_requested_architecture(argc - 2, argv - 2);
```

I would consider changing this logic slightly. This means the first two arguments have been consumed and the caller can just operate on the inputs naturally.